zio/bzngio/index.go

```diff
@@ -0,0 +1,175 @@
+package bzngio
+
+import (
+	"errors"
+	"io"
+	"os"
+
+	"github.com/brimsec/zq/pkg/nano"
+	"github.com/brimsec/zq/zbuf"
+	"github.com/brimsec/zq/zng"
+	"github.com/brimsec/zq/zng/resolver"
+)
+
+type Index struct {
+	index      []mark
+	indexReady bool
+}
+
+type IndexReader interface {
+	zbuf.ReadCloser
+	// count of how many records were read from disk, currently just
+	// used for testing.
+	Reads() uint64
```

Comment:
if this is only used for testing, how about removing it from the public interface and accessing it via a type assertion in the test?

Reply:
something like 

```go
if readser, ok := v.(interface{ reads() uint64 }); ok {
  reads := readser.reads()
 // ...
}
```

Reply:
This is an example where using separate test packages (`package bzngio_test`) isn't helpful. If the test where in `package bzngio`, NewReader could return a zbuf.ReadCloser, and the unit test can type assert directly to `indexReader`.

@aswan : if you just wanted to use `package bzngio` in index_test.go, that would seem ok to me.

Reply:
I don't have any strong opinions here, do you want me to make this change or merge as-is?

Reply:
Make the change!

Reply:
It's ok as-is; apologies for the drive-by comment here. I'll talk to folks about the "_test" package pattern separately.